src/org/opendatakit/briefcase/ui/ScrollingStatusListDialog.java

```diff
@@ -128,8 +130,12 @@ private ScrollingStatusListDialog(Frame frame, String labelText, String title, I
   public void onEvent(FormStatusEvent event) {
     // Since there can be multiple FormStatusEvent's published concurrently,
     // we have to check if the event is meant for this dialog instance.
-    if (event.getStatus().getFormDefinition().equals(form)) {
-      editorArea.setText(event.getStatus().getStatusHistory());
+    if (isShowing() && event.getStatus().getFormDefinition().equals(form)) {
+      try {
+        editorArea.read(new StringReader(event.getStatus().getStatusHistory()), null);
```

Comment:
> I didn't realize that the issue was the total memory usage, I thought it was only the repeated allocation and deallocation of ever larger objects. 

No, you're right. *This* issue is to fix the allocator activity. My point in mentioning the memory usage is that by bounding the buffer you avoid yet another similar memory-bound issue as well.

My apologies in advance for any offense taken to the following, but it is meant only as direct and honest critique of the **code**, not the **authors** and especially not **you**.

If you're *really* careful, keeping around MB-scale log strings in memory is possible as you say. However, it's going to be a MB-scale log multiplied by each form *in memory at the same time* given the current implementation. And for what utility? That's just awful resource usage and for something of questionable value. Keeping a status log in an unbounded buffer, especially given the contents of that log, is at least a questionable decision. That it even has the *opportunity* to threaten the critical functionality of the tool, especially when the user is putting lots of data in it, seems more like a thoughtless design.

There are many ways to offer the same, or even better feedback, without sacrificing critical functionality at scale. One is squelch meaningless status messages to begin with, another is using a more appropriate feedback indicator for progress (like progress bars), another would be just moving the same status to storage and not memory (in other words, using machine resources more appropriately).

An unbounded buffer here is really hard to defend here IMO. 

I don't really have a ton of time, but if the unbounded buffer is a deal-breaker to fix this, I will  resubmit the PR with the unbounded status buffer using a technique similar to what @lognaturel suggested above.